pegasus/sites.v3/code.org/views/congrats_share.haml

```diff
@@ -20,12 +20,36 @@
     trackEvent(social_platform, "share_hoc_certificate", tutorial);
   }
 
+  function replaceShareURLs(newURL) {
+    $('#fb-share-url').attr('href', generateFacebookURL(newURL));
+    $('#twitter-share-url').attr('href', generateTwitterURL(newURL));
+  }
+
+  function generateFacebookURL(urlToShare) {
+    var queryParams = {
+      u: urlToShare
+    };
+
+    return 'https://www.facebook.com/sharer/sharer.php?' + $.param(queryParams, true);
+  }
+
+  function generateTwitterURL(urlToShare) {
+    var queryParams = {
+      url: urlToShare,
+      related: 'codeorg',
+      hashtags: 'HourOfCode', // TODO(bjordan): anon certs may not always be HoC
+      text: "#{@just_did_hoc_text}"
```

Comment:
I'm not super excited about passing the text using HAML interpolation into the JS function but it felt less risky than making assumptions about javascript variable scope / setting order.